pelican/tests/test_readers.py

```diff
@@ -16,10 +16,18 @@ def _path(*args):
 
 class ReaderTest(unittest.TestCase):
 
-    def read_file(self, path, **kwargs):
+    def setUp(self):
+        self._reader = None
+
+    def tearDown(self):
+        self._reader = None
+
+    def read_file(self, path, cache_name='', **kwargs):
         # Isolate from future API changes to readers.read_file
-        r = readers.Readers(settings=get_settings(**kwargs))
-        return r.read_file(base_path=CONTENT_PATH, path=path)
+
+        self._reader = readers.Readers(
+            cache_name=cache_name, settings=get_settings(**kwargs))
```

Comment:
This is technically a side-effect. I think I prefer `reader` created and stored locally in the test, since it's the only one that needs it.